source/developers-guide/plugin-extension-by-plugin/index.md

```diff
@@ -0,0 +1,748 @@
+---
+layout: default
+title: Enhance a plugin with a plugin
+github_link: developers-guide/plugin-extendion-by-plugin/index.md
+indexed: true
+tags:
+  - extend
+  - plugin
+  - custom
+  - products
+---
+
+<div class="toc-list"></div>
+
+## Introduction
+This document will describe how existing plugins can be extended.
+
+With the objective of implement our own option type in CustomProducts which we can use in a Template and make it possible for a customer to upload files with special mime types.
```

Comment:
The objective is to implement an own option type in SwagCustomProducts, which can be used in a template. This new option makes it possible for a customer to upload files with special mime types.